docs: add 'Help Needed' page (#1405)
commercetools/merchant-center-application-kit: website/src/content/help-needed.mdx

---
title: Help needed?
---

<Subtitle>

> In this page we will document some things that might help clarify certain topics or issue.

Reviewer: ```suggestion
This page documents known and recurring challenges with Custom Applications and potential solutions to them.
```

</Subtitle>

<Info>

> Something is not working and you need help? Do you have a technical question, or are you simply looking for some advice?

Reviewer: ```suggestion
Something is not working as expected? You need general guidance? Do you have a technical question or are simply looking for some advice?
```

Author: ```suggestion
Something is not working as expected? Do you need general guidance? Do you have a technical question or are you simply looking for some advice?
```

PS: 
<img width="1390" alt="image" src="https://user-images.githubusercontent.com/1110551/76979480-467e7880-6938-11ea-8187-0b484cc04fc3.png">

Reviewer: Sounds good to me.

> If so, we recommend opening a [GitHub Issue](https://github.com/commercetools/merchant-center-application-kit/issues/new/choose) with any inquire that is of your concern.

Reviewer: ```suggestion
If so, we recommend checking existing [GitHub Issues](https://github.com/commercetools/merchant-center-application-kit/issues) for similar challenges. Given no relating issue exists we recommend opening a new [GitHub Issue](https://github.com/commercetools/merchant-center-application-kit/issues/new/choose) describing your problem.
```

</Info>

# Browser support

> commercetools officially supports the **latest two versions** all major browsers (Google Chrome, Mozilla Firefox, Apple Safari, Microsoft Edge), unless otherwise noted.

Reviewer: We could add a table here maybe and should involve project a bit regarding usage of the MC also I think. 

| [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/edge.png" alt="Edge" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Edge | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/firefox.png" alt="Firefox" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Firefox | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/chrome.png" alt="Chrome" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Chrome | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/safari.png" alt="Safari" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Safari | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/opera.png" alt="Opera" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Opera | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/safari-ios.png" alt="iOS Safari" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>iOS Safari | [<img src="https://raw.githubusercontent.com/godban/browsers-support-badges/master/src/images/chrome-android.png" alt="Chrome for Android" width="16px" height="16px" />](http://godban.github.io/browsers-support-badges/)</br>Chrome for Android |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Edge                                                                                                                                                                                                             | last 2 versions                                                                                                                                                                                                       | last 2 versions                                                                                                                                                                                                    | last 2 versions                                                                                                                                                                                                    | last 2 versions                                                                                                                                                                                                 | last 2 versions                                                                                                                                                                                                                | last version                                                                                                                                                                                                                                       |

Author: I thought about that. I mainly want to make it explicit that we officially support the major browsers. Other browsers might work as well but we don't test on them or something (e.g. opera, mobile browsers, etc).

Not sure if it makes sense to have such a table then

Reviewer: Major browsers is also quite vague I think. The table content was just an idea. We should remove iOS and Opera etc. I copied it from flopflip - I admit that.

The following browsers are not supported any longer:

- `IE 11` and below

<Warning>

It is recommended to always keep your browser up-to-date for **security** reasons.

</Warning>

# Changes to the Custom Application are not reflected immediately

> When a Custom Application is [registered](/register-applications/configuring-a-custom-application), the first time a user visits the Custom Application route within the Merchant Center, the proxy router fetches the configuration of the Custom Application to determine where to proxy the request to.

Reviewer 2: suuuuuper long sentence. Break it up?

Author: ```suggestion
The first time a user visits the Custom Application route in the Merchant Center, the proxy router fetches the configuration of the Custom Application to determine where to proxy the request to.
```

> This configuration is cached within the proxy router to reduce as much as possible the latency introduced by validating each incoming request. The cache is automatically updated at a regular interval (the default value is **30 minutes**).

Reviewer: ```suggestion
This configuration is cached for a certain time within the proxy router to reduce potential latency introduced by validating each incoming request. The cache is automatically updated at a regular interval (the default value is **30 minutes**).
```

> This configuration is cached within the proxy router to reduce as much as possible the latency introduced by validating each incoming request. The cache is automatically updated at a regular interval (the default value is **30 minutes**).

Reviewer 2: ```suggestion
This configuration is cached within the proxy router to reduce the latency introduced by validating each incoming request. The cache is automatically updated at a regular interval (the default value is **30 minutes**).
```

This implies that changes to the Custom Application configuration, for example the **Application URL**, are effectively applied according to the cache interval.

<Info>

> In the future, the cache mechanism will be improved to provide instant cache invalidation whenever something changes in the Custom Application settings.

Reviewer 2: You just publicly committed to deliver something. Sure you want to? Otherwise be more vague:
- we're thinking about improving...
- the mechanism might be improved...

Author: This section "Changes to the Custom Application are not reflected immediately" was already documented in another page, I just moved it here.

But yeah, it's a technical debt which we will fix soon. I'm investigating some options how to do that.

</Info>

# Development login does not work on Chrome

> In Chrome there is a setting option that prevents saving cookies, which makes it impossible to have a valid session when developing the Custom Applications locally. The option **must be turned on**.

Reviewer: ```suggestion
Chrome provides setting which prevents saving any cookies, which makes it impossible to have a valid session when developing the Custom Applications locally. The option **must be turned off**.
```

Author: FYI: The setting must be on, not off. The setting in question is:

```
Allow sites to save and read cookie data (recommended)
```

> In Chrome there is a setting option that prevents saving cookies, which makes it impossible to have a valid session when developing the Custom Applications locally. The option **must be turned on**.

Reviewer 2: ```suggestion
Chrome has a setting option that prevents saving cookies, which makes it impossible to have a valid session when developing the Custom Applications locally. The option **must be turned on**.
```
